tests/backend/checkmach/test_assume_fail.ml

```diff
@@ -58,7 +58,7 @@ let[@zero_alloc strict] test61 x = test60 x
 (* allocations on the path to a call labeled [@zero_alloc assume never_returns_normally]
    do not count for the normal check, but do count for the strict check *)
 
-let[@inline never] test62 x = [x;x]
+let[@inline never] test62 x = if List.length x > 0 then [x;x] else failwith (string_of_int (List.length x))
```

Comment:
I guess the issue is we're combining that assumption (that `test62` never returns normally) with our summary for `test62`.  And I could imagine one of two things are happening:
* We know `test62` always returns normally from its summary, so combined with the knowledge it never returns normally, everything is true, ex falso.
* We know `test62` does not allocate along its exceptional returns (in the original version, where it has none).  So if we know it always returns exceptionally, it definitely doesn't allocate.

OK, I guess whichever of those is happening, it's fine.  But a little surprising.

Reply:
I agree this is rather surprising, but
at the same time we are lying at the
call site. At some point, I thought a
middle ground might be to emit a
warning in such cases, but I now think
it would just shift the problem and
not be more robust in practice.

Reply:
This shouldn't be surprising in common cases in practice because the users would not need to add `[@zero_alloc assume ...]` on the call to `test62` from `test64` if the check was already successful on `test64` without the "assume". The "assume" would be added to strength whatever the analysis inferred about the callee, so using `meet` here is natural.